js/api/Forms.js

```diff
@@ -37,7 +37,27 @@ export function loadCachedFormDataById(formId) {
 
 // Fetches the cached forms related to a specific survey
 export function loadCachedForms(surveyId) {
-  return realm.objects('Form').filtered(`surveyId = "${surveyId}"`)
+  let forms = realm.objects('Form').filtered(`surveyId = "${surveyId}"`)
+  let submissions = realm.objects('Submission')
+  let formList = []
+
+  // Switch out from Realm list to standard array
```

Comment:
Unless I am missing something, realm result is an object and not an array. 
I found excluding/removing target items from it a bit unreliable, as I'd end up with holes in the middle of the list.